### PR TITLE
Aggregation count: Do not return null categories

### PR DESCRIPTION
**Pull request by Algunenano**
Related to https://github.com/CartoDB/support/issues/1368

The issue appeared only if the `aggregationColumn` wasn't provided, and as it's optional in `count ` aggregation, it is being provided in the main map screen but not in the embeds.

### CODE REVIEW

**Comment by Algunenano** [comment]
Ok, seeing https://github.com/CartoDB/Windshaft-cartodb/issues/497 seems that returning a NULL category with 'count' was done in purpose. Needs to be decided then

**Comment by Algunenano** [comment]
The decision is to remove the `NULL` categories in all cases, so I've updated the test that checked it and added some extra test to check that both cases (with and without `aggregationColumn`) don't return NULL values.

File: test/acceptance/dataviews/aggregation.js [hunk 1]
```diff
@@ -425,3 +416,79 @@ describe('aggregation dataview tuned by categories query param', function () {
...
+        this.testClient = new TestClient(mapConfig, 1234);
+        this.testClient.getDataview('categories', { own_filter: 0, categories: 0 }, (err, dataview) => {
+            assert.ifError(err);
+            assert.equal(dataview.categories.length, 3);
+            this.testClient.drain(done);
+        });
+    });
+
+    it(`should handle null values correctly when aggregationColumn is provided`, function (done) {
+        mapConfig.dataviews.categories.options.aggregationColumn = 'val';
```
Comment (oleurud) [inline test/acceptance/dataviews/aggregation.js]: Only a detail. I would prefer to add the aggregation params as a parameter of a function, making future tests independent.

```
function mapConfig (aggregationOptions = {column: 'cat', aggregation: 'count'})
```

File: lib/cartodb/models/dataview/aggregation.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ const rankedCategoriesQueryTpl = ctx => `
             ${ctx.aggregationFn} AS value,
             row_number() OVER (ORDER BY ${ctx.aggregationFn} desc) as rank
         FROM (${filteredQueryTpl(ctx)}) filtered_source
-        ${ctx.aggregationColumn !== null ? `WHERE ${ctx.aggregationColumn} IS NOT NULL` : ''}
+        WHERE ${ctx.aggregation === "count" ? `${ctx.column}` : `${ctx.aggregationColumn}`} IS NOT NULL
```
Comment (oleurud) [inline lib/cartodb/models/dataview/aggregation.js]: :+1:

File: test/acceptance/dataviews/aggregation.js (hunk 1 above)
Comment (Algunenano) [reply test/acceptance/dataviews/aggregation.js]: It's a nice idea, but the only aggregation where this matters is `count` since the rest are invalid if they don't provide an `aggregationColumn`. The others are also tested above.

TBH this test repeats a test above an only the one without `aggregationColumn` is needed, but since the behaviour differed I prefer to have both here together.
